include/nbl/asset/filters/CBlitImageFilter.h

```diff
@@ -520,19 +589,32 @@ class CBlitImageFilter : public CImageFilter<CBlitImageFilter<Normalize,Clamp,Sw
 								for (auto h=0; h<MaxChannels; h++)
 									value[h] += windowSample[h];
 							};
-							// do the filtering 
+							// do the filtering
 							core::vectorSIMDf tmp;
 							tmp[axis] = float(i)+0.5f;
 							core::vectorSIMDi32 windowCoord;
 							windowCoord[axis] = kernel.getWindowMinCoord(tmp*fScale,tmp)[axis];
 							auto relativePos = tmp[axis]-float(windowCoord[axis]);
+							const uint32_t phaseIndex = i % phaseCount[axis];
 							for (auto h=0; h<windowSize; h++)
 							{
-								value_type windowSample[MaxChannels];
+								if (state->enableLUTUsage)
```

Comment:
from now on, you always want to use a LUT

Reply:
you can also get rid of `load` and `evaluate` lambdas declared above.

Reply:
If you do https://github.com/Devsh-Graphics-Programming/Nabla/pull/140/files#r863621098 then you can hoist the `h=0`  case outside and swap the `+=` for a `=` and start the loop at `h=1` and move `windowCoord[axis]++` to be the first line in the loop.

This will allow you to get rid of the `std::fill` at line 627

Reply:
Resolved in #343.